workspaces/adventure-pack/goodies/typescript/Iterator.from/index.test.ts

```diff
@@ -0,0 +1,87 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.from", () => {
+  it("can convert an Array to an Iterator", () => {
+    const array = [1, 2, 3];
+    const iterator = Iterator.from(array);
+
+    expect([...iterator]).toStrictEqual(array);
+  });
+
+  it("can convert a Set to an Iterator", () => {
+    const set = new Set([1, 2, 3]);
+    const iterator = Iterator.from(set);
+
+    expect([...iterator]).toStrictEqual([...set]);
+  });
+
+  it("can convert a Map to an Iterator", () => {
+    const map = new Map([
+      ["a", 1],
+      ["b", 2],
+      ["c", 3],
+    ]);
+    const iterator = Iterator.from(map);
+    expect([...iterator]).toStrictEqual([...map.entries()]);
+  });
+
+  it("can convert a Generator Object to an Iterator", () => {
+    const factory = function* (): Generator<number, void, void> {
+      yield 3;
+      yield 1;
+      yield 4;
+      yield 1;
+      yield 5;
+    };
+    const iterator = Iterator.from(factory());
+
+    expect([...iterator]).toStrictEqual([3, 1, 4, 1, 5]);
+  });
+
+  it("can convert a String to an Iterator", () => {
+    const string = "🚀💯🔥";
+    const iterator = Iterator.from(string);
+
+    expect([...iterator]).toStrictEqual([...string]);
+  });
+
+  it("can convert an Iterator to an Iterator", () => {
+    const array = [1, 2, 3];
+    const iterator = Iterator.from(array.values());
```

Comment:
I'm not sure why we need the double `Iterator.from`, since `array.values()` is an iterator already?

```suggestion
    const iterator = array.values();
```

Reply:
ack!